pkg/pkgmgmt/pkgmgmt.go

```diff
@@ -0,0 +1,51 @@
+package pkgmgmt
+
+import (
+	"os/exec"
+
+	"get.porter.sh/porter/pkg/context"
+)
+
+// PackageManager handles searching, installing and communicating with packages.
+type PackageManager interface {
+	List() ([]Metadata, error)
+	GetSchema(Metadata) (string, error)
+
+	// GetVersion is the obsolete form of retrieving mixin version, e.g. exec version, which returned an unstructured
+	// version string. It will be deprecated soon and is replaced by GetVersionMetadata.
+	GetVersion(Metadata) (string, error)
```

Comment:
Yes I believe we can get rid of it now 👍, all of the mixins support the new signature now. Let's do that in a separate PR from this large refactoring so that it's easier to track down in history later.